oshi-core/src/main/java/oshi/hardware/platform/mac/MacCentralProcessor.java

```diff
@@ -182,9 +182,9 @@ public long[] querySystemCpuLoadTicks() {
     public long[] queryCurrentFreq() {
         if (isArmCpu) {
             Map<Integer, Long> physFreqMap = new HashMap<>();
-            getPhysicalProcessors().stream().forEach(p -> physFreqMap.put(p.getPhysicalProcessorNumber(),
+            getPhysicalProcessors().stream().parallel().forEach(p -> physFreqMap.put(p.getPhysicalProcessorNumber(),
                     p.getEfficiency() > 0 ? performanceCoreFrequency : efficiencyCoreFrequency));
-            return getLogicalProcessors().stream().map(LogicalProcessor::getPhysicalProcessorNumber)
+            return getLogicalProcessors().stream().parallel().map(LogicalProcessor::getPhysicalProcessorNumber)
```

Comment:
NOPE. The `getLogicalProcessors()` is in a particular order and the output of this method has to match.